File: 2656/2656.go
func maximizeSum(nums []int, k int) int {
    max := nums[0]
	for i := 1; i < len(nums); i++ {
		if nums[i] > max {
			max = nums[i]
		}
	}

	result := max
	for i := 1; i < k; i++ {
		result += (max + 1)
		max++
	}

	return result
}

File: 2656/2656.php
class Solution {

    /**
    * @param Integer[] $nums
    * @param Integer $k
    * @return Integer
    */
    function maximizeSum($nums, $k) {
        $max = max($nums);
        $result = $max;
        for ($i = 1; $i < $k; $i++) {
            $result += ++$max;
        }
        return $result;
    }
}